File: python-practice/lesson1_basics.py
# ===========================
# LESSON 1: Python Basics
# ===========================

# --- 1. Variables & Data Types ---
name = "Emmanuel"          # String (text)
age = 25                   # Integer (whole number)
height = 5.9               # Float (decimal number)
is_student = True          # Boolean (True/False)

print("--- Variables ---")
print(f"Name: {name}")
print(f"Age: {age}")
print(f"Height: {height}")
print(f"Student: {is_student}")
print(f"Name type: {type(name)}")
print(f"Age type: {type(age)}")

# --- 2. User Input ---
print("\n--- User Input ---")
favorite_color = input("What is your favorite color? ")
print(f"Cool! {favorite_color} is a great color!")

# --- 3. Math Operations ---
print("\n--- Math ---")
a = 10
b = 3
print(f"{a} + {b} = {a + b}")
print(f"{a} - {b} = {a - b}")
print(f"{a} * {b} = {a * b}")
print(f"{a} / {b} = {a / b}")
print(f"{a} // {b} = {a // b}")   # Floor division
print(f"{a} % {b} = {a % b}")     # Remainder
print(f"{a} ** {b} = {a ** b}")   # Power

# --- 4. If/Else (Conditions) ---
print("\n--- Conditions ---")
score = int(input("Enter your test score (0-100): "))

if score >= 90:
    print("Grade: A - Excellent! 🌟")
elif score >= 80:
    print("Grade: B - Great job! 👍")
elif score >= 70:
    print("Grade: C - Good, keep going!")
elif score >= 60:
    print("Grade: D - You can do better!")
else:
    print("Grade: F - Study harder next time!")

# --- 5. Loops ---
print("\n--- For Loop ---")
for i in range(1, 6):
    print(f"Count: {i}")

print("\n--- While Loop ---")
countdown = 5
while countdown > 0:
    print(f"Countdown: {countdown}")
    countdown -= 1
print("Liftoff! 🚀")

# --- 6. Lists ---
print("\n--- Lists ---")
fruits = ["apple", "banana", "mango", "orange"]
print(f"All fruits: {fruits}")
print(f"First fruit: {fruits[0]}")
print(f"Last fruit: {fruits[-1]}")
fruits.append("grape")
print(f"After adding grape: {fruits}")

print("\n--- Loop through list ---")
for fruit in fruits:
    print(f"I like {fruit}")

print("\n✅ Lesson 1 Complete! You learned:")
print("- Variables & data types")
print("- User input")
print("- Math operations")
print("- If/else conditions")
print("- For & while loops")
print("- Lists")

File: cpp-practice/00-old-files/lesson1_basics.cpp
// ===========================
// LESSON 1: C++ Basics
// ===========================
#include <iostream>
#include <string>
using namespace std;

int main() {

    // --- 1. Variables & Data Types ---
    string name = "Emmanuel";       // Text
    int age = 25;                   // Whole number
    double height = 5.9;            // Decimal number
    char grade = 'A';               // Single character
    bool is_student = true;         // True/False

    cout << "--- Variables ---" << endl;
    cout << "Name: " << name << endl;
    cout << "Age: " << age << endl;
    cout << "Height: " << height << endl;
    cout << "Grade: " << grade << endl;
    cout << "Student: " << is_student << endl;

    // --- 2. User Input ---
    cout << "\n--- User Input ---" << endl;
    string color;
    cout << "What is your favorite color? ";
    cin >> color;
    cout << "Cool! " << color << " is a great color!" << endl;

    // --- 3. Math Operations ---
    cout << "\n--- Math ---" << endl;
    int a = 10, b = 3;
    cout << a << " + " << b << " = " << a + b << endl;
    cout << a << " - " << b << " = " << a - b << endl;
    cout << a << " * " << b << " = " << a * b << endl;
    cout << a << " / " << b << " = " << a / b << " (integer division)" << endl;
    cout << a << " % " << b << " = " << a % b << " (remainder)" << endl;

    // --- 4. If/Else (Conditions) ---
    cout << "\n--- Conditions ---" << endl;
    int score;
    cout << "Enter your test score (0-100): ";
    cin >> score;

    if (score >= 90) {
        cout << "Grade: A - Excellent!" << endl;
    } else if (score >= 80) {
        cout << "Grade: B - Great job!" << endl;
    } else if (score >= 70) {
        cout << "Grade: C - Good, keep going!" << endl;
    } else if (score >= 60) {
        cout << "Grade: D - You can do better!" << endl;
    } else {
        cout << "Grade: F - Study harder next time!" << endl;
    }

    // --- 5. Loops ---
    cout << "\n--- For Loop ---" << endl;
    for (int i = 1; i <= 5; i++) {
        cout << "Count: " << i << endl;
    }

    cout << "\n--- While Loop ---" << endl;
    int countdown = 5;
    while (countdown > 0) {
        cout << "Countdown: " << countdown << endl;
        countdown--;
    }
    cout << "Liftoff!" << endl;

    // --- 6. Arrays ---
    cout << "\n--- Arrays ---" << endl;
    string fruits[] = {"apple", "banana", "mango", "orange"};
    int size = 4;

    cout << "First fruit: " << fruits[0] << endl;
    cout << "Last fruit: " << fruits[size - 1] << endl;

    cout << "\n--- Loop through array ---" << endl;
    for (int i = 0; i < size; i++) {
        cout << "I like " << fruits[i] << endl;
    }

    cout << "\n--- Lesson 1 Complete! You learned: ---" << endl;
    cout << "- Variables & data types" << endl;
    cout << "- User input (cin)" << endl;
    cout << "- Math operations" << endl;
    cout << "- If/else conditions" << endl;
    cout << "- For & while loops" << endl;
    cout << "- Arrays" << endl;

    return 0;
}
